### ISSUE TITLE
Register PS Repository fails with 'SourceLocation' is an invalid Web Uri

### ISSUE THREAD

**Issue by papanito** [PowerShell/PowerShell#5708]
Registering of a private powershell gallery fails with `'SourceLocation' is an invalid Web Uri.` even so the URI seems correct and corresponds to the one provided by my Nexus3 server.

- I have direct access so no proxy
- Changing the url from https to http gives the same error
- I can successfully browse the repository

Steps to reproduce
------------------

```powershell
Register-PSRepository -Name TestRepo -SourceLocation https://nexus3/repository/nuget-group -PublishLocation https://nexus3/repository/nuget-psgallery/ -PackageManagementProvider nuget -InstallationPolicy Trusted
```

Expected behavior
-----------------
PS Repository is registered successfully

Actual behavior
---------------
Error is thrown
```powershell
Register-PSRepository : The specified Uri 'https://nexus3/repository/nuget-group' for parameter
'SourceLocation' is an invalid Web Uri. Please ensure that it meets the Web Uri requirements.
At line:1 char:1
+ Register-PSRepository -Name TestRepo -SourceLocation https://nexus ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (https://nexus..ory/nuget-group:String) [Register-PSRepository], Arg
   umentException
    + FullyQualifiedErrorId : InvalidWebUri,Register-PSRepository
```

Environment data
----------------
```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.14393.1884
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.1884
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

>Get-PackageProvider

Name                     Version          DynamicOptions
----                     -------          --------------
msi                      3.0.0.0          AdditionalArguments
msu                      3.0.0.0
NuGet                    2.8.5.208        Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag...
PowerShellGet            1.0.0.1          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, ...
Programs                 3.0.0.0          IncludeWindowsInstaller, IncludeSystemComponent

```

**Comment by iSazonov** [comment]
@papanito Could you repo with latest PowerShell Core build?

**Comment by bergmeister** [comment]
Same error for ` v6.1.0-preview.518` but that is maybe because the given Uri cannot even be pinged:
````powershell
ping https://nexus3/repository/nuget-group
Ping request could not find host https://nexus3/repository/nuget-group. Please check the name and try again.
````
Sometimes I feel it would've been better if this repo was named `PowerShellCore` or something similar but I understand that it is probably too late now.  But I can also understand that people rather post here because `UserVoice` is frankly speaking a dead letter box for most MSFT products (I do the same for VSTS issues tbh.). What about having at least a separate `WindowsPowerShell` repo just for issues that concern only `Windows PowerShell` (similar to how [WiX](https://github.com/wixtoolset/issues) does it)? @SteveL-MSFT

**Comment by papanito** [comment]
@iSazonov I can't update my PS. More interestingly, I could do the same command successfully last Friday on the same host. Still try to figure out what is different from Friday to Monday ....

@bergmeister you have to ping your host e.g. `ping nexus3` not `ping https://nexus3`

**Comment by iSazonov** [comment]
@papanito PowerShell Core don't replace Windows PowerShell. You can safely install it side-by-side (and just simply copy-paste in any folder from zip distro).

**Comment by papanito** [comment]
Same problem with PowerShell-6.0.0-rc.2-win-x64

**Comment by papanito** [comment]
Ok probably not a but at all.... When I enter the url from source location into the browser I get the following error from Nexus3:
```
Repository path must have another '/' after initial '/'
```

So after adding a trailing / to the url, no more error

```
 https://nexus3/repository/nuget-group/
````

**Comment by iSazonov** [comment]
@markekraus Could you please comment the Issue?

**Comment by markekraus** [comment]
@iSazonov I'm not sure what help I can be. The PowerShellGet module and Register-PSRepository don't use the Web Cmdlets. It directly calls to either System.Net.Http.HttpClient or System.Net.WebRequest. the InvalidWebUri error is thrown for a bunch fo different reason in that module. 

Only thing I can suggest to @papanito  is to use the server's FQDN instead of just `https://nexus3/`. But I'm no expert on this module.

**Comment by iSazonov** [comment]
@markekraus Thanks! Maybe the question is for PowerShellGet repository.

**Comment by bergmeister** [comment]
I agree, maybe @rebro-msft  can help out?

**Comment by rebro-msft** [comment]
I believe this issue is already being tracked in PowerShellGet repo: [Issue 180](https://github.com/PowerShell/PowerShellGet/issues/180)
And has a [ Stack Overflow post](https://stackoverflow.com/questions/35296482/invalid-web-uri-error-on-register-psrepository).

I'm not familiar with nexus repository, so not sure if the following applies to your scenario.  But when I saw something similar (registering a debug session of PowerShell Gallery locally) it was complaining because the SSL cert used by HTTPS was not in the Trusted Root.  When I fixed that, the issue resolved.

Another workaround is to  I registered the http endpoint, then edit the repositories xml file to use the https endpoint instead.
